### PR TITLE
Dm 3738

### PR DESCRIPTION
**Pull request by JoshingYou1**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-3738

## Description - what does this code do?
Address issues with certain facilities in the `va_facilities.json` having no complexity and being duplicates of facilities that already exist.

## Testing done - how did you test it/steps on how can another person can test it 
1. Open up a rails console: `rails c`, and run the following script: 
```
VaFacility.where(fy17_parent_station_complexity_level: '').destroy_all
```
2. That should have deleted the duplicate facilities in your DB.
3. Now visit `/diffusion-map` and make sure the map loads with no errors. Also, in the filters dropdown, make sure all complexity level checkboxes are formatted as "Level type" (Level description), e.g., 1A (High complexity).
4. Now, update the existing Stockton VA Clinic facility with the data from the json. Run the following script:
```
VaFacility.find_by(slug: 'stockton').update(sta3n: '612', station_number: '612QE', classification: 'Other Outpatient Services (OOS)', classification_status: 'Firm', mobile: 'No', parent_station_number: '612A4', official_parent_station_name: 'Sacramento VA Medical Center', fy17_parent_station_complexity_level: '1b-High Complexity', operational_date: '7/1/2022', date_of_first_workload: '10/1/2021', after_hours_phone_number: '[800-382-8387](tel:8003828387) x', pharmacy_phone_number: '[866-600-8279](tel:8666008279) x', enrollment_coordinator_phone_number: '[916-843-7225](tel:9168437225) x', patient_advocate_phone_number: '[916-843-7034](tel:9168437034) x', va_secretary_10n_approved_date: '5/20/2022', planned_activation_date: '7/1/2022', station_number_suffix_reservation_effective_date: '7/1/2022', latitude: 37.88266367.to_f, longitude: -121.2819955.to_f)
```
5. Go to any facility show page and make sure the word "veterans" in the blurb below the name of the facility is now capitalized.
6. Run `rails va_facilities:create_or_update_va_facilities`, visit `/diffusion-map`, `/facilities`, and any other facility-related pages to make sure everything still functions as intended.

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Unit tests written (if applicable)
- [ ] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [X] A link has been provided to the originating JIRA issue
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/views/va_facilities/_index_table_row.html.erb [hunk 1]
```diff
@@ -16,7 +16,10 @@
     <% if facility.class.name === "ClinicalResourceHub" %>
         <td class="grid-col-2">-</td>
      <% else %>
-        <td class="grid-col-2"><%= facility.fy17_parent_station_complexity_level.match(/^1/) ? facility.fy17_parent_station_complexity_level[0..1].upcase : facility.fy17_parent_station_complexity_level[0..0].upcase %></td>
+        <td class="grid-col-2">
+          <% complexity = facility.fy17_parent_station_complexity_level %>
+          <%= complexity.match(/^1/) ? complexity[0..1].upcase : complexity[0..0].upcase %>
+        </td>
```
Comment (JoshingYou1) [inline app/views/va_facilities/_index_table_row.html.erb:22]: Cleaned up code a bit.

File: app/views/va_facilities/show.html.erb [hunk 2]
```diff
@@ -31,8 +31,9 @@
           <% when "I" %>
             isolated
         <% end %>
+        <% facility_opening_date = @va_facility.station_number_suffix_reservation_effective_date %>
         <%= @va_facility.fy17_parent_station_complexity_level %>
-        facility and serves veterans in <%= @va_facility.mailing_address_city %>.  <% unless @va_facility.station_number_suffix_reservation_effective_date.blank? %>This facility opened in <%= @va_facility.station_number_suffix_reservation_effective_date.to_s[0..3] %>.<% end %>
+        facility and serves Veterans in <%= @va_facility.mailing_address_city %>.<%= " This facility opened in #{facility_opening_date.to_s[0..3]}." if facility_opening_date.present? %>
```
Comment (JoshingYou1) [inline app/views/va_facilities/show.html.erb:36]: Cleaned up code a bit.

File: lib/tasks/va_facilities.rake [hunk 3]
```diff
@@ -8,24 +8,24 @@ namespace :va_facilities do
...
+                  station_number: vaf["Station Number"].to_s,
                   official_station_name: vaf["Official Station Name"],
                   common_name: vaf["Location Descriptive Name"],
                   classification: classification,
                   classification_status: vaf["ClassificationStatus"],
                   mobile: vaf["Mobile"],
-                  parent_station_number: vaf["Official Parent Station Number"],
-                  official_parent_station_name: vaf["Parent Station Name"],
+                  parent_station_number: vaf["Official Parent Station Number"].to_s,
+                  official_parent_station_name: vaf["Official Parent Station Name"],
```
Comment (JoshingYou1) [inline lib/tasks/va_facilities.rake:26]: We were looking for the wrong key name in the json

**Review by bradjohnson92008 — CHANGES_REQUESTED** [review]
So it looks like the console commands and rails create_or_update task all ran correctly.. but I"m still seeing the issue with loading the maps - seeing this javascript loading without a call back... and reference ot the deprecated addDomListener method... 
image

**Review by camillevilla — APPROVED** [review]
Took this for a spin in local with master rebased on top. Things are working great and I definitely see some of the cleanup in the views logic. Great work! Let's get this deployed tomorrow.

**Pull request merged**